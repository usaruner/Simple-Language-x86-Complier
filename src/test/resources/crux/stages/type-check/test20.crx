
// test statementBlock

var num:int;

func main() : void {
  var five:int;
  var sum:int;

  let num = 2;
  let five = 5;
  let sum = num + five;
}
