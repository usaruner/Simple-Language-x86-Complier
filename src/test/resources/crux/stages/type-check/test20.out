Crux Program has no type errors.
ast.DeclarationList(4)
  ast.VariableDeclaration(4)[Symbol(num:int)]
  ast.FunctionDefinition(6)[Symbol(main:func(TypeList()):void), []]
    ast.StatementList(7)
      ast.VariableDeclaration(7)[Symbol(five:int)]
      ast.VariableDeclaration(8)[Symbol(sum:int)]
      void: ast.Assignment(10)
        Address(int): ast.Name(10)[num]
        int: ast.LiteralInt(10)[2]
      void: ast.Assignment(11)
        Address(int): ast.Name(11)[five]
        int: ast.LiteralInt(11)[5]
      void: ast.Assignment(12)
        Address(int): ast.Name(12)[sum]
        int: ast.OpExpr(12)[ADD]
          int: ast.Dereference(12)
            Address(int): ast.Name(12)[num]
          int: ast.Dereference(12)
            Address(int): ast.Name(12)[five]
